desktop/core/src/desktop/js/utils/hueUtils.ts

```diff
@@ -530,93 +523,118 @@ export const getFileBrowseButton = (
-              inputElement.attr('data-original-title', inputElement.val());
-              const _val = inputElement.val();
-              inputElement.val(_val.split('/')[_val.split('/').length - 1]);
+        if (
+          typeof valueAccessor() === 'function' ||
+          typeof (<ComplexValueAccessor>valueAccessor)().value === 'function'
+        ) {
+          const complex = (<ComplexValueAccessor>valueAccessor)();
+          if (complex.value) {
+            complex.value(<string>inputElement.val());
+            if (complex.displayJustLastBit) {
+              inputElement.data('fullPath', <string>inputElement.val());
+              inputElement.attr('data-original-title', <string>inputElement.val());
+              const value = <string>inputElement.val();
+              inputElement.val(value.split('/')[value.split('/').length - 1]);
             }
-          } else {
-            valueAccessor()(inputElement.val());
+            return;
           }
-        } else {
-          valueAccessor(inputElement.val());
         }
+        (<KnockoutObservable<string>>valueAccessor())(<string>inputElement.val());
       }
-    }
+    };
   });
   if (allBindingsAccessor && allBindingsAccessor().filechooserDisabled) {
-    _btn.addClass('disabled').attr('disabled', 'disabled');
+    button.addClass('disabled').attr('disabled', 'disabled');
   }
-  return _btn;
+  return button;
 };
 
-const stripHtml = html => {
+const stripHtml = (html: string): string => {
   const tmp = document.createElement('DIV');
   tmp.innerHTML = html;
   return tmp.textContent || tmp.innerText;
 };
 
-export const stripHtmlFromFunctions = template => {
+export const stripHtmlFromFunctions = (template: string): string => {
   // strips HTML from inside the functions
-  let _tmpl = template;
-  const _mustacheFunctions = _tmpl.match(/{{#(.[\s\S]*?){{\//g);
-  if (_mustacheFunctions) {
-    $.each(_mustacheFunctions, (cnt, fn) => {
-      _tmpl = _tmpl.replace(
+  let stripped = template;
+  const mustacheFunctions = stripped.match(/{{#(.[\s\S]*?){{\//g);
+  if (mustacheFunctions) {
+    mustacheFunctions.forEach(fn => {
+      stripped = stripped.replace(
         fn,
         fn.substr(0, fn.indexOf('}}') + 2) +
-          $.trim(stripHtml(fn.substr(fn.indexOf('}}') + 2).slice(0, -3))) +
+          stripHtml(fn.substr(fn.indexOf('}}') + 2).slice(0, -3)).trim() +
           '{{/'
       );
     });
   }
-  return _tmpl;
+  return stripped;
 };
 
-export const sleep = async timeout => new Promise(resolve => setTimeout(resolve, timeout));
+export const sleep = async (timeout: number): Promise<void> =>
+  new Promise(resolve => setTimeout(resolve, timeout));
 
-export const defer = async callback => {
+export const defer = async (callback: () => void): Promise<void> => {
   await sleep(0);
   if (callback) {
     callback();
   }
 };
 
+export const hueLocalStorage = <T = string>(key: string, value?: T): T | null => {
```

Comment:
Nit: As hueUtils is crowded, we could probably move this to storageUtils or something similar. 
And in this case using a generic name without the word **Local** would enable us to switch to a different WebApi like IndexDB without much code change in the future.

Reply:
Indeed a separate makes sense! There are currently a bunch of js outside the webpack modules and hueUtils is accessible through window.hueUtils for those. I didn't want to crowd the global object more than necessary for this change but to keep in mind indeed.

There's a distinct difference between localStorage and IndexedDB, where IndexedDB has async operations and localStorage not so swapping here has implications. Could be good to know that we also use IndexedDB through localForage, the dataCatalog caching for instance.

Reply:
I'll move this out in separate commit.